Ira/my-app-video/src/components/home/Home.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import {RouteComponentProps} from "react-router-dom";
+import {Header} from "../layout/header";
+import {Footer} from "../layout/footer";
+import {FilmList} from "../filmList";
+import {SearchResultSection} from "../searchResultSection";
+import SearchFilmForm from "../shared/searchFilmForm";
+import {IFilm} from "../../interfaces/IFilm";
+import {queryString} from "../../constants/queryString";
+import './styles.scss';
+
+interface IOwnProps {
+  result: number,
+  films: IFilm[],
+
+  showFullFilmInfo(id: number): () => void,
+
+}
+
+interface IHomeState {
+  sortBy: string,
+}
+
+type HomeProps = RouteComponentProps & IOwnProps;
+
+export class Home extends React.Component<HomeProps, IHomeState> {
+  state: IHomeState = {
+    sortBy: 'release',
+  };
+
+  componentDidMount() {
+    const locationSearch = this.props.location.search;
+    const params = queryString.parse(locationSearch);
+
+    this.setState({
+      sortBy: params.sortBy || 'release',
+    });
+  }
+
+  componentDidUpdate(prevProps: Readonly<HomeProps>) {
+    if (this.props.history.action !== 'PUSH' && this.props.location !== prevProps.location) {
+      const params = queryString.parse(this.props.location.search) as { search: string, searchBy: string, sortBy: string };
+      params.sortBy = params.sortBy || 'release';
+
+      this.setState({sortBy: params.sortBy});
+    }
+  };
+
+  onSearchSubmit = ({searchValue, searchBy}: { searchValue: string, searchBy: string }): void => {
+
+    const pathname: string = `/search?search=${searchValue}&searchBy=${searchBy}`;
```

Comment:
why didn't you do like  you did in **onToggleSortBy** ? You can't just replace an URL, you have to keep sortBy param there.